vision_agent/agent/vision_agent_v2.py

```diff
@@ -167,6 +167,7 @@ def write_and_exec_code(
     retrieved_ltm: str,
     max_retry: int = 3,
     verbosity: int = 0,
+    log_progress: Callable[..., str] = None,
```

Comment:
to fix the type checking error, consider change to below type:
```suggestion
    log_progress: Optional[Callable[..., str]] = None,
```